core/shared/src/main/scala/zio/internal/FiberContext.scala

```diff
@@ -30,15 +30,14 @@ import scala.annotation.{switch, tailrec}
  */
 private[zio] final class FiberContext[E, A](
   protected val fiberId: Fiber.Id,
-  platform: Platform,
+  var platform: Platform,
   startEnv: AnyRef,
   startExec: Executor,
   startLocked: Boolean,
   startIStatus: InterruptStatus,
   parentTrace: Option[ZTrace],
   initialTracingStatus: Boolean,
   var fiberRefLocals: FiberRefLocals,
-  supervisor0: Supervisor[Any],
   openScope: ZScope.Open[Exit[E, A]],
   reportFailure: Cause[Any] => Unit
```

Comment:
In a similar vein, I might get rid of `startExec` (use the one in platform, modifying as appropriate), and `reportFailure` (use the one in platform, modifying as appropriate).

Reply:
I'm a little worried about semantics here. `startExec` means "the executor we should start running on" which in `fork` I don't think is necessarily the same as "the default executor to run on if none is specified". In particular I think that means if we did `ZIO.platform.fork.flatMap(_.join)` we might not get the same value as `ZIO.platform` which seems strange. There is a little bit of the same issue in `reportFailure`. For example `race` uses `noop` for `reportFailure` so the platform in the branches of a race would be different than the platform in the parent.

Reply:
You are right about this 👍